### PR TITLE
#1337 isolation level fix for all values accept string

### PR DESCRIPTION
**Pull request by enesacikoglu**

### CODE REVIEW

File: src/main/java/com/zaxxer/hikari/util/UtilityElf.java [hunk 1]
```diff
@@ -165,30 +164,14 @@ public static ThreadPoolExecutor createThreadPoolExecutor(final BlockingQueue<Ru
...
-               case Connection.TRANSACTION_SERIALIZABLE:
-               case Connection.TRANSACTION_NONE:
-               case SQL_SERVER_SNAPSHOT_ISOLATION_LEVEL: // a specific isolation level for SQL server only
-                  return level;
-               default:
-                  throw new IllegalArgumentException();
-             }
-         }
-         catch (Exception e) {
+            final String upperCaseIsolationLevelName = Optional.of(transactionIsolationName.toUpperCase(Locale.ENGLISH))
```
Comment (turbanoff) [inline src/main/java/com/zaxxer/hikari/util/UtilityElf.java]: Why `Optional` is used here? It's absolutely redundant

Comment (enesacikoglu) [reply src/main/java/com/zaxxer/hikari/util/UtilityElf.java]: Right,please check last commit.

**Comment by enesacikoglu** [comment]
Hello,
Waiting for review.
Thanks.

**Comment by sadikkuzu** [comment]
Congrats @enesacikoglu

**Comment by brettwooldridge** [comment]
@enesacikoglu It looks like this change actually causes a regression.  Previous versions of HikariCP could accept an integer value, such as 4096, as the isolation level.  After this change, only strings of the form "TRANSACTION*" are supported, therefore existing users who were using integer values will now be broken.

It is my fault for not catching this in my review of the code.

**Comment by enesacikoglu** [comment]
> @enesacikoglu It looks like this change actually causes a regression. Previous versions of HikariCP could accept an integer value, such as 4096, as the isolation level. After this change, only strings of the form "TRANSACTION*" are supported, therefore existing users who were using integer values will now be broken.
> 
> It is my fault for not catching this in my review of the code.

Hello @brettwooldridge ,

Actually, I was aware of doing this change because it seems to be not configured correctly. It must be the string instead of number because client can not understand which isolation level refers to that number.

I know , it is a kind of breaking change but I would not want to add a workaround solution inside code for only 4096.

If it is a fault, it is not only your fault.

Finally , we can document this change as a breaking change for the client before to build a new version. I hope it will work.

Thanks
Enes

**Pull request merged**